component/src/main/java/io/siddhi/extension/io/file/FileHandler.java

```diff
@@ -0,0 +1,322 @@
+import io.siddhi.core.stream.ServiceDeploymentInfo;
+import io.siddhi.core.stream.input.source.Source;
+import io.siddhi.core.stream.input.source.SourceEventListener;
+import io.siddhi.core.util.config.ConfigReader;
+import io.siddhi.core.util.snapshot.state.State;
+import io.siddhi.core.util.snapshot.state.StateFactory;
+import io.siddhi.core.util.transport.OptionHolder;
+import io.siddhi.extension.io.file.util.Constants;
+import io.siddhi.extension.io.file.util.Status;
+import io.siddhi.extension.util.Utils;
+import io.siddhi.query.api.exception.SiddhiAppValidationException;
+import org.apache.commons.io.monitor.FileAlterationMonitor;
+import org.apache.commons.io.monitor.FileAlterationObserver;
+import org.apache.commons.vfs2.FileObject;
+import org.apache.commons.vfs2.FileSystemException;
+import org.apache.log4j.Logger;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static io.siddhi.extension.io.file.util.Util.getFileHandlerEvent;
+/**
+ * Implementation of siddhi-io-file Event Listener.
+ * FileEventListener  provides the functionality for user to get the details of files which have been created,
+ * modified or deleted in the execution time.
+ */
+@Extension(
+        name = "fileeventlistener" ,
+        namespace = "source" ,
+        description = "" +
+                "Fileeventlistener provides the functionality for user to get the details of files " +
+                "which have been created or modified or deleted in the execution time." ,
+        parameters = {
+                @Parameter(
+                        name = "dir.uri",
+                        description =
+                                "This parameter is used to specify a folder to be processed. " +
+                                        "All the files inside this directory will be processed. " +
+                                        "This uri MUST have the respective protocol specified.",
+                        optional = false,
+                        type = {DataType.STRING}
+                ),
+                @Parameter(
+                        name = "monitoring.interval",
+                        description =
+                                "This parameter is used to specify the time interval (in milliseconds) " +
+                                        "that the process monitor the changes for.\n",
+                        type = {DataType.STRING},
+                        optional = true,
+                        defaultValue = "100"
+                ),
+                @Parameter(
+                        name = "file.name.list",
+                        description = "This parameter is used to filter the files to be monitored in the" +
+                                " given directory uri (dir.uri). The files should be entered comma separated eg:" +
+                                " 'abc.txt,xyz.csv'\n",
```

Comment:
@Ajanthy theres an additional "\n" added. shall we remove this since it goes to the documentation?

Reply:
fixed in https://github.com/siddhi-io/siddhi-io-file/pull/82/commits/3181f69f399c713c78213b5b3e2cfef0d7442373